.dir-locals.el

```diff
@@ -0,0 +1,23 @@
+;; This causes emacs to abide by formatting rules: No tab characters, except
```

Comment:
Should this be added to the /highlight/emacs/ directory?

Reply:
Other people have been using emacs on Chapel code from the beginning without requiring this file. I wonder if these would just be considered personal settings?

Reply:
> Should this be added to the /highlight/emacs/ directory?

I don't know what `/highlight` is. This is an `emacs` directory setting which applies to all `emacs` users, so that the formatting rules which are enforced by the CI tests are automatically observed by `emacs` with minimal fuss.

Reply:
> Other people have been using emacs on Chapel code from the beginning without requiring this file. I wonder if these would just be considered personal settings?

You can certainly put it in your `~/.emacs` file, or put `.dir-locals.el` in your home directory,  but since these formatting rules are enforced by the Chapel CI, these settings to automatically use these formatting rules should be provided to all `emacs` users, without them having to do any special configuration just to edit Chapel.

Reply:
In other words, if your personal `emacs` preferences are to always prefer tabs instead of spaces, and to use an indentation width of 8, you should not have to edit your preferences just to edit Chapel source, keep in line with its style, and pass the CI tests. Chapel source code can enforce formatting rules at the source directory level, and when you edit files under that directory, you will use those formatting rules instead of your default `emacs` preferences.